front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx

```diff
@@ -1,49 +1,210 @@
+  };
+
+  useEffect(() => {
+    const checkTruncation = () => {
+      setIsTruncated((prev) => ({
+        scenarioName: scenarioNameRef.current
+          ? scenarioNameRef.current.scrollWidth > scenarioNameRef.current.clientWidth
+          : prev.scenarioName,
+        username: usernameRef.current
+          ? usernameRef.current.scrollWidth > usernameRef.current.clientWidth
+          : prev.username,
+      }));
+    };
+    checkTruncation();
+    window.addEventListener('resize', checkTruncation);
+    return () => {
+      window.removeEventListener('resize', checkTruncation);
+    };
+  }, []);
+
   return (
-    <header className="scenario-header">
-      {/* scenario info */}
-      <div className="scenario-info">
-        <button className="close-btn" type="button">
-          <X />
-        </button>
-
-        <span className="scenario-name">{scenarioName}</span>
-
-        <button className="chevron-btn" type="button">
-          <ChevronDown />
-        </button>
-      </div>
+    <header className="scenario-header-container">
+      <div className="scenario-header">
+        {/* scenario info */}
+        <div className="scenario-info">
+          <button className="close-btn" type="button" onClick={closeScenario}>
+            <X />
+          </button>
+
+          <div className="inactive-area" />
 
-      {/* board display management */}
-      <nav className="board-bar">
-        {boards.map((board) => (
-          <button key={board} className="board-btn" type="button">
-            {board}
+          <span className="close-label">{t('translation:common.close')}</span>
+
+          <div
+            className="scenario-name-container"
+            role="button"
+            tabIndex={0}
+            onClick={toggleScenarioDetails}
+          >
+            <span
+              ref={scenarioNameRef}
+              className={cx('scenario-name-label', { 'is-truncated': isTruncated.scenarioName })}
+            >
+              {scenario.name}
+            </span>
+
+            <button className="chevron-btn" type="button">
+              {areScenarioDetailsVisible ? <ChevronUp /> : <ChevronDown />}
+            </button>
+          </div>
+
+          <div className="spacer" />
+        </div>
+
+        {/* board display management */}
+        <nav className="board-bar">
+          <div className="spacer" />
+
+          <div className="board-btns">
+            {boards.map((board, index) => (
+              <Fragment key={board}>
+                <button
+                  className={cx('board-btn', {
+                    on: activeBoards.includes(board),
+                  })}
+                  type="button"
+                  onClick={() => {
+                    toggleBoard(board);
+                  }}
+                >
+                  {board}
+                </button>
+                {index < boards.length - 1 && <div className="inactive-area" />}
+              </Fragment>
+            ))}
```

Comment:
~Can you do that ? I think you can even remove the fragment if you do that~
```suggestion
              </Fragment>
            ))}
            {index < boards.length - 1 && <div className="inactive-area" />}
            
```

Reply:
Not relevant, sorry !